src/mgmt/srv_drpc.c

```diff
@@ -66,6 +66,43 @@ pack_daos_response(Mgmt__DaosResp *daos_resp, Drpc__Response *drpc_resp)
 	drpc_resp->body.data = body;
 }
 
+void
+ds_mgmt_drpc_prep_shutdown(Drpc__Call *drpc_req, Drpc__Response *drpc_resp)
+{
+	Mgmt__PrepShutdownReq	 *req = NULL;
+	Mgmt__DaosResp		 *resp = NULL;
+
+	/* Unpack the inner request from the drpc call body */
+	req = mgmt__prep_shutdown_req__unpack(
+		NULL, drpc_req->body.len, drpc_req->body.data);
+	if (req == NULL) {
+		drpc_resp->status = DRPC__STATUS__FAILURE;
```

Comment:
Just a note, you'll have to change this status code once #1544 lands.

Reply:
https://github.com/daos-stack/daos/pull/1591